chain-impl-mockchain/src/config.rs

```diff
@@ -822,78 +792,23 @@ impl ConfigParamVariant for CommitteeId {
 }
 
 #[cfg(feature = "evm")]
-impl ConfigParamVariant for Box<EvmConfigParams> {
+impl ConfigParamVariant for EvmConfig {
     fn to_payload(&self) -> Vec<u8> {
-        let bb: ByteBuilder<EvmConfigParams> = ByteBuilder::new().u8(self.config as u8);
-        let env = &self.environment;
-        let bb = bb
-            .bytes(&<[u8; 32]>::from(env.gas_price))
-            .bytes(env.origin.as_fixed_bytes())
-            .bytes(&<[u8; 32]>::from(env.chain_id));
-        let bb = if env.block_hashes.is_empty() {
-            bb.u64(0)
-        } else {
-            let bb = bb.u64(env.block_hashes.len().try_into().unwrap());
-            env.block_hashes
-                .iter()
-                .fold(bb, |b, h| b.bytes(h.as_fixed_bytes()))
-        };
-        let bb = bb
-            .bytes(&<[u8; 32]>::from(env.block_number))
-            .bytes(env.block_coinbase.as_fixed_bytes())
-            .bytes(&<[u8; 32]>::from(env.block_timestamp))
-            .bytes(&<[u8; 32]>::from(env.block_difficulty))
-            .bytes(&<[u8; 32]>::from(env.block_gas_limit))
-            .bytes(&<[u8; 32]>::from(env.block_base_fee_per_gas));
+        let bb: ByteBuilder<EvmConfig> = ByteBuilder::new().u8(*self as u8);
         bb.finalize_as_vec()
     }
 
     fn from_payload(payload: &[u8]) -> Result<Self, Error> {
         let mut rb = ReadBuf::from(payload);
 
         // Read EvmConfig and match hard fork variant
-        use EvmConfig::*;
         let config = match rb.get_u8()? {
-            n if n == Istanbul as u8 => Istanbul,
-            n if n == Berlin as u8 => Berlin,
+            n if n == EvmConfig::Istanbul as u8 => EvmConfig::Istanbul,
+            n if n == EvmConfig::Berlin as u8 => EvmConfig::Berlin,
```

Comment:
Can it be transformed into just this?

```rust
        let config = match rb.get_u8()? {
            EvmConfig::Istanbul as u8 => EvmConfig::Istanbul,
            EvmConfig::Berlin as u8 => EvmConfig::Berlin,
            _ => return Err(Error::InvalidTag),
        };
```

Reply:
I have tried, unfortunately not possible